hashintel/hash
# Support `null`, `emptyList` & `object` data type in entity editor

## 🌟 What is the purpose of this PR?

This PR adds support for `null`, `emptyList` and `object` types in entity editor. So instead of seeing the fallback `TextEditor`, users will see proper UI for these three types.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203179076056209/1203715515231598/f) _(internal)_

## 🔍 What does this change?

- Add `prismjs` to highlight JSON in `json-editor`
- Create a const for glide-grid util classname `click-outside-ignore`
- Rename `entityFromDB` -> `entityFromDb`
- Update single-value and array editor logics to support new types
- Move `editorSpecs` to it's own file since it's getting bigger with more editors
- Refactor render code of `sortable-row`
- add `arrayValueToString` util function to render a preview of arrays on grid cells
- improve `guessEditorTypeFromExpectedType`, now it searches for the given type in `editorSpecs`

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Create an entity type with `object` in it (if you want to test `emptyList` and `null` types, you need to use the API to create properties from those types manually via code.)
2. Confirm that new object editor & property table UI/UX works as expected

## 📹 Demo

https://user-images.githubusercontent.com/39553853/214262235-29e37f49-5e20-4f2b-b3f3-91e761726fbd.mov
